jenkins/Jenkinsfile-blossom.premerge

```diff
@@ -118,16 +119,27 @@ pipeline {
                         def CUDA_NAME = sh(returnStdout: true,
                                 script: '. jenkins/version-def.sh>&2 && echo -n $CUDA_CLASSIFIER | sed "s/-/./g"')
 
-                        IMAGE_TAG = "dev-ubuntu16-${CUDA_NAME}"
-                        CACHE_IMAGE_NAME = "${ARTIFACTORY_NAME}/sw-spark-docker/plugin:${IMAGE_TAG}"
+                        // check if pre-merge dockerfile modified
+                        def dockerfileModified = sh(returnStdout: true,
+                                script: 'BASE=$(git --no-pager log --oneline -1 | awk \'{ print $NF }\'); ' +
```

Comment:
I don't think this does what we want.  Printing `$NF` is going to print the last word on the line which will be the last word in the comment one-liner, but that's not going to be useful for obtaining a commit hash.

I think we just want to use `git rev-parse HEAD` here, no `awk` processing necessary.

Reply:
@jlowe `git --no-pager log --oneline -1 | awk '{ print $NF }'` is trying to get the base commit for diff comparing, and the HEAD ref is used directly in `git merge-base HEAD $BASE`.

In our pre-merge cases, we fetch the PRs from remote `+refs/pull/*/merge:refs/remotes/origin/pr/*`, and we used the merged code (merge HEAD to BASE branch) to run the test to avoid people using out-dated code (but no merge conflict) to pass the CI.

So the HEAD commit message is always like `Merge HEAD into BASE`, e.g. `Merge 586f28f20dbe1621c743722627766e4439cf0014 into dc66f03a4664f41a2f16ddef4a587f81a3fc41c0`. I don't know if we have a better way to get the BASE commit w/o fetching the remote BASE branch, and I don't think `git rev-parse HEAD or origin/branch-0.5` would work here. Please let me know if any better option, thanks!